### PR TITLE
Consider draft pull requests to be dev_blocked

### LINKED ISSUE #230: Move draft pull requests into its own column
Having draft pull requests in Pulldasher clutters it up. Let's move them so that they don't show up in other columns.

## Do this
Treat Draft pulls the same as "dev_blocked" pulls. They should stay in the "Dev Block" column till they're ready for review.

### PR DESCRIPTION
**Pull request by jordycosta**
### Overview:

Previously, a pull request marked as a draft would stay in the **CR/QA** column. This pull request will adjust that by placing these _draft-state_ pull requests into the **Dev blocked** column.

draft

> May want to add a sort to this eventually

### CR/QA Notes:

> **CR:** Review to https://github.com/iFixit/pulldasher/pull/329/commits/6063432395f411dd4fccaabfc7323d035f43563b

> **QA:**
>
> You can probably just skip to step 3.)
>
> 1. Checkout | merge with [`draft-dev-block`](https://github.com/iFixit/pulldasher/tree/draft-dev-block)
> 2. Run `/home/ifixit/ifixit_scripts/update-pulldasher-dev --run .`
> 3. Navigate to https://pulldasher-dev.cominor.com/
> 4. Ensure that draft pull requests have a flag and are placed in the **Dev blocked** column

---

cr_req 2

Closes #230

### CODE REVIEW

**Comment by jordycosta** [comment]
dev_block 🟥

**Review by andyg0808 — APPROVED** [review]
This is looking pretty good! I think we'll need to look at how to get the `draft` status into the DB before we can ship it, but so far it looks good.

**Comment by andyg0808** [comment]
CR :dog: through c2b780f
deploy_block :+1: on potentially handing DB as well.

File: frontend/src/pull-card/flags.tsx [hunk 1]
```diff
@@ -32,6 +33,11 @@ function Flags({pull}: {pull: Pull}) {
          href={pull.linkToSignature(devBlock)}
          icon={faMinusCircle}
       />}
+      {draftBlock && <PullFlag
+         variant="draftBlock"
+         title={"Draft"}
+         icon={faMinusCircle}
```
Comment (andyg0808) [inline frontend/src/pull-card/flags.tsx:39]: Is this supposed to be `faMinusSquare`? If not, you might want to drop the `faMinusSquare` above.

File: frontend/src/pulldasher/index.tsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ export const Pulldasher: React.FC = function() {
    const pullsCIBlocked = pulls.filter(pull => pull.isCiBlocked());
    const pullsDeployBlocked = pulls.filter(pull => pull.isDeployBlocked());
    const pullsReady = pulls.filter(pull => pull.isReady() && pull.isCiRequired());
-   const pullsDevBlocked = pulls.filter(pull => pull.getDevBlock());
+   const pullsDevBlocked = pulls.filter(pull => pull.getDevBlock() && pull.isDraft());
    const pullsNeedingCR = pulls.filter(pull => !pull.isCrDone() && !pull.getDevBlock());
    const pullsNeedingQA = pulls.filter(pull => !pull.isQaDone() && !pull.getDevBlock() && pull.hasPassedCI());
```
Comment (andyg0808) [inline frontend/src/pulldasher/index.tsx:17]: I think you'll need to add a `!isDraft` check to the CR and QA pulls lists:
```suggestion
   const pullsNeedingCR = pulls.filter(pull => !pull.isCrDone() && !pull.getDevBlock() && !pull.isDraft());
   const pullsNeedingQA = pulls.filter(pull => !pull.isQaDone() && !pull.getDevBlock() && pull.hasPassedCI() && !pull.isDraft());
```

File: models/db_pull.js [hunk 3]
```diff
@@ -12,6 +12,7 @@ function DBPull(pull) {
       state: pullData.state,
       title: pullData.title,
       body: pullData.body,
+      draft: pullData.draft,
```
Comment (andyg0808) [inline models/db_pull.js:15]: I think you'll also need to add this column to the schema. Looks like we normally add a migration in the `migrations` directory, and it seems we also update the `schema.sql` file (not sure if that's manual, or automatic somehow):
https://github.com/iFixit/pulldasher/tree/draft-dev-block/migrations

Here's a commit that seems to be doing the thing:
https://github.com/iFixit/pulldasher/commit/ae18be0a17f7c7d5896bd6aca1c7a76b1f0205b0

**Comment by andyg0808** [comment]
CR :dog: through b9fdd1a
deploy_block :+1: on the DB schema change.

File: frontend/src/pull-card/flags.tsx (hunk 1 above)
Comment (jordycosta) [reply frontend/src/pull-card/flags.tsx:39]: I forgot to update the icon value, thank you!

Comment (jordycosta) [reply frontend/src/pull-card/flags.tsx]: Resolved via https://github.com/iFixit/pulldasher/pull/329/commits/c3af2d9290bc2aa0cd95e35380937afc411fe2f0

File: models/db_pull.js (hunk 3 above)
Comment (danielbeardsley) [reply models/db_pull.js:15]: It's manual.

Comment (jordycosta) [reply models/db_pull.js:15]: Updated schema.sql file via https://github.com/iFixit/pulldasher/pull/329/commits/03023cbd41d23d33d7814906697fd9957a137496

File: migrations/schema.sql [hunk 4]
```diff
@@ -130,6 +130,7 @@ CREATE TABLE IF NOT EXISTS `pulls` (
   `state` enum('open','closed') COLLATE utf8mb4_general_ci NOT NULL,
   `title` varchar(255) COLLATE utf8mb4_general_ci NOT NULL,
   `body` text COLLATE utf8mb4_general_ci NOT NULL,
+  `draft` boolean default false,
```
Comment (danielbeardsley) [inline migrations/schema.sql:133]: We also need a migration file to make this mutation. Use the `AFTER {column}` syntax too.

Comment (jordycosta) [reply migrations/schema.sql]: Added migration file via https://github.com/iFixit/pulldasher/pull/329/commits/b6cbe56787129ad52255d4b7cac2fac5f9c5b8dd

File: models/db_pull.js (hunk 3 above)
Comment (jordycosta) [reply models/db_pull.js]: Column added via https://github.com/iFixit/pulldasher/pull/329/commits/1fc5deb414751b1a79f666cb4cd41074bae04326

File: models/db_pull.js [hunk 5]
```diff
@@ -12,7 +12,7 @@ function DBPull(pull) {
       state: pullData.state,
       title: pullData.title,
       body: pullData.body,
-      draft: pullData.draft,
+      draft: pullData.draft === 1,
```
Comment (danielbeardsley) [inline models/db_pull.js:15]: The comment here says this takes in API data and should return DB data. So I'd expect this to be `pullData.draft ? 1 : 0` instead.

Comment (jordycosta) [reply models/db_pull.js]: Resolved via https://github.com/iFixit/pulldasher/pull/329/commits/54ae9a6d38c471c2e3bd45a0dfd5c288f4357fec

**Comment by danielbeardsley** [comment]
CR :+1: 

Looks great! Good work @jordycosta !

**Comment by danielbeardsley** [comment]
deploy_block :+1:  on applying this migration in prod.

File: frontend/src/pulldasher/index.tsx (hunk 2 above)
Comment (jordycosta) [reply frontend/src/pulldasher/index.tsx]: Resolved via https://github.com/iFixit/pulldasher/pull/329/commits/79bc61452c623139e34e13c4a9f006b8144c2283

**Comment by danielbeardsley** [comment]
CR :+1:

**Comment by jordycosta** [comment]
un_deploy_block 🟩 

>https://github.com/iFixit/ifixit/pull/44108 merged

**Comment by rickisXP** [comment]
QA :page_facing_up: 
- PR with draft status are now filtered into the `dev_block` column :heavy_check_mark: 
- New draft icon is visible and spacing appears fine :heavy_check_mark:

File: models/pull.js [hunk 6]
```diff
@@ -220,6 +221,7 @@ Pull.getFromDB = function(data, signatures, comments, reviews, commitStatuses, l
       state: data.state,
       title: data.title,
       body: data.body,
+      draft: data.draft === 1,
```
Comment (andyg0808) [inline models/pull.js:224]: Just a JS tip: `0` is `false` in JS, and non-zero numbers are `true`. So you could use

```suggestion
      draft: Boolean(data.draft),
```

or traditionally

```js
      draft: !!data.draft,
```
(`!` results in a boolean value; double-negation keeps the boolean the same; it's a idiomatic trick)

**Review by andyg0808 — APPROVED** [review]

**Comment by andyg0808** [comment]
CR :dog: through 6063432

File: models/pull.js (hunk 6 above)
Comment (danielbeardsley) [reply models/pull.js:224]: True, I advised the `=== 1` to be really clear what data was expected from the DB.

**Pull request merged**